architecture/libraries.rst

```diff
@@ -4,24 +4,100 @@
 Libraries
 #########
 
+.. _libutils:
+
+libutils
+********
+
+OP-TEE core and OP-TEE development kit for Trusted Application provide a
+standard C library that is named **libutils**. It implements many
+standard functions like ``snprintf()``, ``strncmp()``, ``memcpy()``,
+``malloc()``. ``qsort()``, and many more but not all standard C library
+functions.
+
+Note however that Trusted Applications implemented in C should use GP TEE
+Internal Core API functions rather than their standard C library function
+equivalent (e.g. ``TEE_MemMove()`` instead of ``memcpy()`` and
+``memmove()``, or ``TEE_Malloc()`` instead of ``malloc()`` and friends).
+This makes those TAs implementation more portable to other GP
+TEE compliant environments.
+
+When ``CFG_ULIBS_SHARED`` is enabled, **libutils** is assigned UUID
+**71855bba-6055-4293-a63f-b0963a737360**.
+
 .. _libutee:
 
 libutee
 *******
 The :ref:`tee_internal_core_api` describes services that are provided to Trusted
 Applications. **libutee** is a library that implements this API.
 
-libutee is a static library the Trusted Applications shall statically link
-against. Trusted Applications do execute in non-privileged secure userspace and
-libutee also aims at being executed in the non-privileged secure userspace.
+libutee is designed as a userland library specifically dedicated to OP-TEE
+Trusted Applications and aims at being executed in the non-privileged secure
+userspace.
 
 Some services for this API are fully statically implemented inside the libutee
 library while some services for the API are implemented inside the OP-TEE core
 (privileged level) and libutee calls such services through system calls.
 
-.. _libmpa:
+When ``CFG_ULIBS_SHARED`` is enabled, **libutee** is assigned UUID
+**4b3d937e-d57e-418b-8673-1c04f2420226**.
+
+libmbedtls
+**********
+
+OP-TEE OS source tree provides support of the Mbed TLS library, named
+**libmbedtls**.
+
+A specific build sequence compiles an instance of **libmbedtls** and links
+it to OP-TEE. Another build sequence compiles an instance of **libmbedtls**
+that will be linked with Trusted Applications.
+
+When Mbed TLS is embedded in OP-TEE core, it is used as the default software
+implementation for cryptography operations. When so, **libtomcrypt** must
+be explicitly disabled are ther are exclusive as default software crypto
```

Comment:
This is not entirely accurate. When using Mbed TLS we still rely on LTC for some algorithms.

Reply:
**are ther are**  ... too many **are**? and **ther** misspelled.

Reply:
@jbech-linaro, thanks. Check the latest description that rephrases this sentence.